client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java

```diff
@@ -447,6 +448,20 @@ public final BulkByScrollResponse reindex(ReindexRequest reindexRequest, Request
         );
     }
 
+    /**
+     * Submits a reindex task.
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html">Reindex API on elastic.co</a>
+     * @param reindexRequest the request
+     * @param options the request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
+     * @return the submission response
+     * @throws IOException in case there is a problem sending the request or parsing back the response
+     */
+    public final ReindexSubmissionResponse submitReindexTask(ReindexRequest reindexRequest, RequestOptions options) throws IOException {
+        return performRequestAndParseEntity(
```

Comment:
I wonder if there's any value to returning a ReindexSubmissionResponse rather than just a TaskId here?

Reply:
Yeah, maybe as TaskId is good here.

Reply:
I was considering this, but the reason I stayed with a Wrapping object is that Rest API returns taskId wrapped inside an object where TaskID is just one field. Same is in  DeleteJobResponse (ML).
However I don't mind breaking this convention. It would however look a bit odd inside TaskID to have a parser that treats taskId as a field within itself. See TaskID from latest revision where I did that.

Reply:
Hmmmmmmm. I see. So the idea is that if we added more than as `TaskId` to the response then we could stick it on this object. We can do that in the REST response because of the way the JSON is structured. The way your have it we could do that here. I like that argument. Even though it *looks* like an extra argument, it makes sense. It might be worth sticking on the Javadoc of that object. Somewhere.

Reply:
Im +1 for a response object matching the response name (like ReindexSubmissionResponse) for this "adding things to the API" reason as well.

Reply:
I was thinking of staying with TaskSubmissionResponse with String taskId field. As this would fit into other wait_for_completion=false type methods as well. 

We could in theory always make the taskSubmission abstract and return concrete responses - with all the parsing and fields being already in the abstract class. That would be a bit too complex for that I guess

Reply:
I'm fine with `TaskSubmissionResponse`. And I'm fine with `String taskId`. I'd also be fine with a client-side `TaskId` class that is super opaque. It is an identifier that client's aren't meant to pick apart. A `String` is fine for that though.